edge_orchestrator/src/edge_orchestrator/utils/usb_device.py

```diff
@@ -0,0 +1,49 @@
+import logging
+from typing import Dict, Optional, Tuple
+
+from pyudev import Context
+
+
+def list_connected_usb_device() -> Dict[Tuple, str]:
+    context = Context()
+    usb_devices = {}
+    for device in context.list_devices(subsystem="video4linux"):
+        if device.get("ID_BUS", "Unknown") == "usb":
+            logging.debug(f"Found USB device: {device}")
+            vendor_id = device.get("ID_VENDOR_ID", None)
+            product_id = device.get("ID_MODEL_ID", None)
+            device_path = device.get("DEVNAME", None)
+            vendor_name = device.get("ID_VENDOR", "Unknown")
+            model_name = device.get("ID_MODEL", "Unknown")
+            if vendor_id and product_id and device_path and (vendor_id.strip(), product_id.strip()) not in usb_devices:
+                usb_devices[(vendor_id.strip(), product_id.strip())] = {
+                    "vendor_name": vendor_name,
+                    "model_name": model_name,
+                    "device_path": device_path,
+                }
+                logging.debug(
+                    f"Vendor id: {vendor_id.strip()}, Product id: {product_id.strip()}, Device Path: {device_path}, Camera Name: {model_name}, Vendor Name: {vendor_name}"
+                )
+    return usb_devices
+
+
+def get_camera_device_node(camera_vendor: str, camera_serial_number: str) -> Optional[str]:
```

Comment:
We can add a parameter representing the number of streams the camera has and a parameter to indicate how many exactly same camera are connected to the system. For instance, the UP camera use 3 or 4 mounting points like this exemple:
- camera UP 1:
  - /dev/video0
  - /dev/video1
  - /dev/video2 
  - /dev/video3
- camera UP 2:
  - /dev/video4
  - /dev/video5
  - /dev/video6 
  - /dev/video7

We could use this parameter (nb of streams=4) combine to the number of exact same camera (=2) to return the first mounting point of the camera for the first camera. In this case /dev/video0. For the second camera, it would return /dev/video4.